pyspectral/rsr_reader.py

```diff
@@ -126,7 +127,7 @@ def __init__(self, platform_name=None, instrument=None, **kwargs):
         super(RelativeSpectralResponse, self).__init__()
 
         self.platform_name = platform_name
-        self.instrument = instrument
+        self.instrument = AVHRR_INSTRUMENT_NAME.get(instrument, instrument)
```

Comment:
This seems like an odd place to have something AVHRR specific. Why isn't this handled in `_check_instrument`? Could `_check_instrument` return a new converted instrument instead of nothing like it does now? Oh I see `_check_instrument` is only called if instrument and platform are both provided...that should still work right?

Reply:
You are right, thanks!
Done some refactoring, also got rid of the `**kwargs` in the class initialization.